jsonToPrometheus/collector/prometheusCollector.go

```diff
@@ -0,0 +1,195 @@
+// collected by this Collector to the provided channel and returns once
+// the last descriptor has been sent. The sent descriptors fulfill the
+// consistency and uniqueness requirements described in the Desc
+// documentation.
+func (collector *prometheusCollector) Describe(descriptorsChan chan<- *prometheus.Desc) {
+	descriptorsChan <- collector.descriptors.tempRating
+	descriptorsChan <- collector.descriptors.numLeaderSuccess
+	descriptorsChan <- collector.descriptors.numLeaderFailure
+	descriptorsChan <- collector.descriptors.numValidatorSuccess
+	descriptorsChan <- collector.descriptors.numValidatorFailure
+	descriptorsChan <- collector.descriptors.numValidatorIgnoredSignatures
+	descriptorsChan <- collector.descriptors.rating
+	descriptorsChan <- collector.descriptors.ratingModifier
+	descriptorsChan <- collector.descriptors.totalNumLeaderSuccess
+	descriptorsChan <- collector.descriptors.totalNumLeaderFailure
+	descriptorsChan <- collector.descriptors.totalNumValidatorSuccess
+	descriptorsChan <- collector.descriptors.totalNumValidatorFailure
+	descriptorsChan <- collector.descriptors.totalNumValidatorIgnoredSignatures
+	descriptorsChan <- collector.descriptors.shardId
+	descriptorsChan <- collector.descriptors.validatorStatus
+}
+
+// Collect is called by the Prometheus registry when collecting
+// metrics. The implementation sends each collected metric via the
+// provided channel and returns once the last metric has been sent. The
+// descriptor of each sent metric is one of those returned by Describe
+// (unless the Collector is unchecked, see above). Returned metrics that
+// share the same descriptor must differ in their variable label
+// values.
+func (collector *prometheusCollector) Collect(metricsChan chan<- prometheus.Metric) {
+	validatorStatistics, err := collector.httpClient.GetValidatorStatistics(context.Background())
+	if err != nil {
+		log.Error("could not fetch validator statistics", "error", err.Error())
+		return
+	}
+
+	if len(collector.blsKeys) == 0 {
+		log.Info("no bls key provided, will serve metrics for all keys available")
+		collector.collectMetricsForAllKeys(metricsChan, validatorStatistics)
+		return
+	}
+
+	collector.collectMetricsForManagedKeys(metricsChan, validatorStatistics)
+
+}
+
+func (collector *prometheusCollector) collectMetricsForAllKeys(
+	metricsChan chan<- prometheus.Metric,
+	validatorStatistics map[string]*httpClientWrapper.ValidatorStatistics,
+) {
+	for blsKey, statistics := range validatorStatistics {
+		collector.collectMetricsForKey(metricsChan, statistics, blsKey)
+	}
+}
+
+func (collector *prometheusCollector) collectMetricsForManagedKeys(
+	metricsChan chan<- prometheus.Metric,
+	validatorStatistics map[string]*httpClientWrapper.ValidatorStatistics,
+) {
+	for _, blsKey := range collector.blsKeys {
+		statistics, found := validatorStatistics[blsKey]
+		if !found {
+			log.Error("no validator statistics found", "key", blsKey)
```

Comment:
I do not think this log error is necessary. Reason: we can start the monitoring while the node is in the staking queue or not-staked and this tool will fill up the logs with errors.

Reply:
removed it